File: apps/android/feature/analytics/domain/src/main/java/com/scribblefit/feature/analytics/domain/repository/AnalysisRepository.kt
package com.scribblefit.feature.analytics.domain.repository

import com.scribblefit.feature.ai.domain.model.AnalysisSuggestion
import com.scribblefit.feature.ai.domain.model.AnalysisSummary
import com.scribblefit.feature.ai.domain.model.ExerciseInsight
import com.scribblefit.feature.ai.domain.model.SummaryPeriod
import kotlinx.coroutines.flow.Flow

interface AnalysisRepository {
    fun getHomeSuggestion(): Flow<AnalysisSuggestion?>
    fun getSummary(period: SummaryPeriod): Flow<AnalysisSummary?>
    fun getExerciseInsight(exerciseId: String): Flow<ExerciseInsight?>

    suspend fun saveHomeSuggestion(suggestion: AnalysisSuggestion)
    suspend fun saveSummary(summary: AnalysisSummary)
    suspend fun saveExerciseInsight(insight: ExerciseInsight)
    suspend fun clearOldInsights()
}

File: apps/ios/ScribbleFit/Sources/Features/Analytics/Domain/Repository/AnalysisRepository.swift
import Combine
import Foundation

public protocol AnalysisRepository: Sendable {
    func getHomeSuggestion() -> AnyPublisher<AnalysisSuggestion?, Never>
    func getSummary(period: SummaryPeriod) async throws -> AnalysisSummary?
    func getExerciseInsight(exerciseId: String) async throws -> ExerciseInsight?
    func saveHomeSuggestion(_ suggestion: AnalysisSuggestion) async throws
    func saveSummary(_ summary: AnalysisSummary) async throws
    func saveExerciseInsight(_ insight: ExerciseInsight) async throws
    func clearOldInsights() async throws
}
